test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -85,13 +89,29 @@ session_simulator::rollback_transaction()
     _txn_running = false;
 }
 
-void
-session_simulator::commit_transaction()
+int
+session_simulator::commit_transaction(const std::string &config)
 {
     /* Make sure that the transaction from this session is running. */
     assert(_txn_running);
 
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+    std::map<std::string, std::string> config_map;
+
+    ts_manager->parse_config(config, config_map);
+
+    auto pos = config_map.find("commit_timestamp");
+    if (pos != config_map.end()) {
+        uint64_t commit_ts = ts_manager->hex_to_decimal(pos->second);
```

Comment:
Is there any kind of verification that would be worth doing on `pos->second`? Can it have bad values or something `hex_to_decimal` would not interpret correctly?

Reply:
Trying a non hex value such as "g" results in the Illegal commit timestamp: zero not permitted." below so I will add a more appropriate message in that case.

Reply:
Yes, indicating the value that leads to a failure could be great.

Reply:
```suggestion
    auto pos = config_map.find("commit_timestamp");
    if (pos != config_map.end()) {
        for (auto &ch : pos->second)
            if (!std::isxdigit(ch))
                WT_SIM_RET_MSG(EINVAL, "Illegal commit timestamp: invalid hex value.");

        uint64_t commit_ts = ts_manager->hex_to_decimal(pos->second);
        if (commit_ts == 0)
            WT_SIM_RET_MSG(EINVAL, "Illegal commit timestamp: zero not permitted.");
        config_map.erase(pos);
        WT_SIM_RET(set_commit_timestamp(commit_ts));
    }
```

Reply:
Would this be an okay approach?

Reply:
Do we need to look at each character? [This example](https://cplusplus.com/reference/cctype/isxdigit/) does not seem to do it.

Reply:
I'm not sure if I'm missing something but that example only appears to be checking the first character? What if it's "fggg"?

Reply:
Yep, my bad I did not look properly at the example.